libs/components/theme/src/lib/providers/provide-modern-theme.ts

```diff
@@ -0,0 +1,44 @@
+import { DOCUMENT } from '@angular/common';
+import {
+  APP_INITIALIZER,
+  EnvironmentProviders,
+  RendererFactory2,
+  makeEnvironmentProviders,
+} from '@angular/core';
+
+import { SkyTheme } from '../theming/theme';
+import { SkyThemeMode } from '../theming/theme-mode';
+import { SkyThemeSettings } from '../theming/theme-settings';
+import { SkyThemeSpacing } from '../theming/theme-spacing';
+import { SkyThemeService } from '../theming/theme.service';
+
+export function provideModernTheme(options?: {
```

Comment:
I propose making this more generic by taking in a parameter for the theme name (`default` or `modern`) and also using a name that indicates this sets the initial theme since the theme can change during the application's lifecycle (e.g. the user selects dark mode or compact mode). Perhaps `provideInitialTheme()` or `provideThemeInit()`?